llvm/include/llvm/CodeGen/BasicTTIImpl.h

```diff
@@ -2164,6 +2164,53 @@ class BasicTTIImplBase : public TargetTransformInfoImplCRTPBase<T> {
       // Otherwise, fallback to default scalarization cost.
       break;
     }
+    case Intrinsic::loop_dependence_raw_mask:
```

Comment:
I think you should be able to move this to an independent PR, and also test this with dedicated cost-model tests.